### DISCUSSION TITLE
Does logPurchase support item arrays with multiple objects?

### DISCUSSION THREAD

**Discussion by chipoglesby** [invertase/react-native-firebase#6891] [Q&A]
I have been able to successfully use `logPurchase` with a single item object that is sent to Firebase/GA4 for collection and analysis.

However, I am unsuccesful at sending an item array with multiple objects. I have been unable to find a solution online or in the documentation.

Any suggestions or help you may have would be greatly appreciated.

 ### This works
```
item = {
  item_id "xxx_1",
  item_name: "xxx_1",
  Item_quantity: 2,
  index: 1,
  price: 12.99,
  item_category: "xxx",
};
```

### This does not work
```
item = [item_1, item_2]

item_1 = {
  item_id "xxx_1",
  item_name: "xxx_1",
  Item_quantity: 2,
  index: 1,
  price: 12.99,
  item_category: "xxx"
};

item_2 = {
  item_id "xxx_2",
  item_name: "xxx_2",
  Item_quantity: 2,
  index: 2,
  price: 12.99,
  item_category: "xxx",
};

**Comment by mikehardy** [comment]
With apologies, I'm unsure - this hasn't come up before. Have you asked on stackoverflow with appropriate firebase tagging? The firebase team answers there, and if firebase-js-sdk can do it, and/or either of the native platforms, we *should* be able to

**Reply by chipoglesby** [reply]
@mikehardy, I have, but I have not found a solution _**yet**_.

Firebase supports a [parcelable array for items](https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Param#ITEMS). 

GA4 [supports it as well](https://developers.google.com/analytics/devguides/collection/ga4/reference/events?client_type=gtag#purchase).
